pypeit/specobj.py

```diff
@@ -350,39 +350,67 @@ def copy(self):
+        """
+        Perform interpolation on wave given a shift in pixels
+
+        Args:
+            shift (float):
+                Shift in pixels
+            wave (`numpy.ndarray`_):
+                extracted wave of size nspec
+            wavein (`numpy.ndarray`_, optional):
+                Apply the shift to this array of wavelengths
+        Returns:
+            `numpy.ndarray`_: Wavelength scale corrected for spectral flexure
+
+        """
+        npix = wave.size
+        x = np.linspace(0., 1., npix)
+        f = interpolate.interp1d(x, wave, bounds_error=False, fill_value="extrapolate")
+        twave = f(x + shift / (npix - 1))
+        # If this shift is to be applied onto another wave scale, reinterpolate
+        if wavein is not None:
+            f = interpolate.interp1d(wave, twave, bounds_error=False, fill_value="extrapolate")
+            twave = f(wavein)
+        return twave
+
+    def apply_spectral_flexure(self, fdict):
         """
         Apply interpolation with the flexure dict
 
         Args:
-            sky_wave (np.ndarray): Wavelengths of the extracted sky
-            fdict (dict): Holds the various flexure items
+            fdict (dict):
+                Holds the various flexure items
 
         Returns:
             xspectrum1d.XSpectrum1D:  New sky spectrum (mainly for QA)
-
         """
         # Simple interpolation to apply
-        npix = len(sky_wave)
-        x = np.linspace(0., 1., npix)
         # Apply
         for attr in ['BOX', 'OPT']:
             if self[attr+'_WAVE'] is not None:
                 msgs.info("Applying flexure correction to {0:s} extraction for object:".format(attr) +
                           msgs.newline() + "{0:s}".format(str(self.NAME)))
-                f = interpolate.interp1d(x, sky_wave, bounds_error=False, fill_value="extrapolate")
-                self[attr+'_WAVE'] = f(x + fdict['shift'] / (npix - 1)) #* units.AA
+                self[attr+'_WAVE'] = self.flexure_interp(fdict['shift'], self[attr+'_WAVE']).copy()
         # Shift sky spec too
-        cut_sky = fdict['sky_spec']
-        x = np.linspace(0., 1., cut_sky.npix)
-        f = interpolate.interp1d(x, cut_sky.wavelength.value, bounds_error=False, fill_value="extrapolate")
-        twave = f(x + fdict['shift'] / (cut_sky.npix - 1)) * units.AA
-        new_sky = xspectrum1d.XSpectrum1D.from_tuple((twave, cut_sky.flux))
-        # Save
-        self.FLEX_SHIFT = fdict['shift']
+        twave = self.flexure_interp(fdict['shift'], fdict['sky_spec'].wavelength.value) * units.AA
+        new_sky = xspectrum1d.XSpectrum1D.from_tuple((twave, fdict['sky_spec'].flux))
+        # Save - since flexure may have been applied/calculated twice, this needs to be additive
+        self.update_flex_shift(fdict['shift'])
         # Return
         return new_sky
 
+    def update_flex_shift(self, shift):
+        """Store the total spectral flexure shift in pixels
+
```

Comment:
I'm not following what we are summing here. Perhaps this is for when we have done both slit and object based flexure shifts, but perhaps we should be storing those shifts individually then as well.

Reply:
That's right... this is the total shift. If you prefer, we can store a `local` and a `global` spectral flexure?

Reply:
Yeah, I think storing local, global, and the sum of the flexure is a good thing to do.